pytensor/tensor/rewriting/linalg.py

```diff
@@ -972,20 +969,55 @@ def rewrite_cholesky_diag_to_sqrt_diag(fgraph, node):
     return [eye_input * (non_eye_input**0.5)]
 
 
-@node_rewriter([_bilinear_solve_discrete_lyapunov])
-def jax_bilinaer_lyapunov_to_direct(fgraph: FunctionGraph, node: Apply):
-    """
-    Replace BilinearSolveDiscreteLyapunov with a direct computation that is supported by JAX
-    """
-    A, B = (cast(TensorVariable, x) for x in node.inputs)
-    result = solve_discrete_lyapunov(A, B, method="direct")
+def _check_log_abs_det(fgraph, client):
+    # First, we find abs
+    if not (isinstance(client.op, Elemwise) and isinstance(client.op.scalar_op, Abs)):
+        return False
 
-    return [result]
+    # Check whether log is a client of abs
+    for client_2 in fgraph.clients[client.outputs[0]]:
+        if not (
+            isinstance(client_2[0].op, Elemwise)
+            and isinstance(client_2[0].op.scalar_op, Log)
+        ):
+            return False
 
+    return True
 
-optdb.register(
-    "jax_bilinaer_lyapunov_to_direct",
-    in2out(jax_bilinaer_lyapunov_to_direct),
-    "jax",
-    position=0.9,  # Run before canonicalization
-)
+
+@register_specialize
+@node_rewriter([det])
+def slogdet_specialization(fgraph, node):
+    replacements = {}
+    for client in fgraph.clients[node.outputs[0]]:
+        # Check for sign(det)
+        if isinstance(client[0].op, Elemwise) and isinstance(
+            client[0].op.scalar_op, Sign
+        ):
+            x = node.inputs[0]
+            sign_det_x, slog_det_x = SLogDet()(x)
+            replacements[client[0].outputs[0]] = sign_det_x
```

Comment:
If there are multiple replacements would be nice to avoid creating the SlodDet multiple times. Can we rewrite this more elegantly perhaps store a list of variables to be replaced and what they are and then create replacements in a second loop? 

```python
dummy_replacements= {}
for client in ...:
  if var is sign:
    dummy_replacements[var] = "sign"
  elif:
    ...

if not replacements:
  return 

[x] = node.inputs
sign, log_abs_det = SlogDet()(x)
log_det = switch(...)
replacements = {k: (sign if v == ”sign” else log_abs_det if v == "log_abs_det” else log_det) for k, v in dummy_replacements.items()}
```